lib/internal/modules/cjs/loader.js

```diff
@@ -1674,6 +1675,7 @@ function getRequireESMError(mod, pkg, content, filename) {
   const usesEsm = containsModuleSyntax(content, filename);
   const err = new ERR_REQUIRE_ESM(filename, usesEsm, parentPath,
                                   packageJsonPath);
+  ErrorCaptureStackTrace(err, Module.prototype.require);
```

Comment:
The whole stack - including internals - is something like 

```
at getRequireESMError (node:internal/modules/cjs/loader:1683:15)
    at Object..js (node:internal/modules/cjs/loader:1724:17)
    at Module.load (node:internal/modules/cjs/loader:1310:32)
    at Function._load (node:internal/modules/cjs/loader:1124:12)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at blah (node:internal/modules/cjs/loader:210:22)
    at wrapModuleLoad (node:internal/modules/cjs/loader:229:12)
    at Module.require (node:internal/modules/cjs/loader:1333:12)
    at require (node:internal/modules/helpers:136:16)
    at Object.<anonymous> (/Users/edysilva/projects/contributions/node/test/fixtures/es-modules/package-type-module/require-esm-error-annotation/index.cjs:1:13) {
  code: 'ERR_REQUIRE_ESM'
  ```
  
  We would then need to strip from `require` and above.

Reply:
Using the `hideStackFrames` would work too. I don't know the possible impacts of doing that though. That's why I used `ErrorCaptureStackTrace`.

Let me know if `hideStackFrames` can be used instead.

cc @RafaelGSS

Reply:
I tried to put the `hideStackFrames` around `wrapModuleLoad`. This does not work because, for some reason, the internal errors are not filtered out. Even though [hideInternalStackFrames](https://github.com/nodejs/node/blob/main/lib/internal/errors.js#L1662) is there.